src/bindings/python/src/pyopenvino/core/common.cpp

```diff
@@ -193,7 +194,11 @@ void fill_tensor_from_strings(ov::Tensor& tensor, py::array& array) {
             PyUnicode_FromKindAndData(PyUnicode_4BYTE_KIND, reinterpret_cast<void*>(ptr), buf.itemsize / 4);
         PyObject* _utf8_obj = PyUnicode_AsUTF8String(_unicode_obj);
         const char* _tmp_str = PyBytes_AsString(_utf8_obj);
-        data[i] = std::string(_tmp_str);
+        if (!Common::utils::is_string_null_terminated(_tmp_str)) {
+            data[i] = std::string(_tmp_str);
+        } else {
+            data[i] = std::string(_tmp_str, buf.ndim == 0 ? buf.itemsize : buf.strides[0]);
```

Comment:
Can we always use this constructor?
I guess we always have string length

Reply:
Using this constructor breaks other Python tests, for example numerous cases from `test_runtime/test_tensor_string.py`. I'm currently working on figuring out why.